### ISSUE TITLE
Parsing a table name that starts with number returns an error

### ISSUE THREAD

**Issue by sljohnsondev** [AlaSQL/alasql#1185]
When passing a basic SELECT statement to the parser with a table name that starts with a number (e.g. '50ks'), I am incorrectly getting an error.  This is not the case if the number is later in the table name (e.g. 'ks50').

Request: "SELECT * FROM csv.50k"
Error msg: "Parsing halted while starting to recover from another error."

If I add backticks around the table name, it works, but wanted to flag b/c I believe this should be considered valid as well.  Please correct me if that is wrong!

In either case, it would be great to get a better error message aligned with other errors from the parser.  

When I dug into the parser, I noticed that a better error message is getting flagged but is getting overwritten before the actual error is thrown.  _That msg is - "Parse error on line 1:\nSELECT * FROM csv.50k\n-----------------^\nExpecting 'LITERAL', 'BRALITERAL', 'EOF', 'WITH', 'COMMA', 'AS', 'LPAR', 'RPAR', 'PIVOT', 'UNPIVOT', 'DOT', 'ORDER', 'WHERE', 'NOT', 'UNION', 'INTERSECT', 'EXCEPT', 'FROM', 'CROSS', 'OUTER', 'NATURAL', 'JOIN', 'INNER', 'LEFT', 'RIGHT', 'FULL', 'SEMI', 'ANTI', 'ON', 'USING', 'GROUP', 'LIMIT', 'OFFSET', 'END', 'ELSE', 'GO', 'SEMICOLON', got 'NUMBER'"_

**Comment by mathiasrw** [comment]
Good point. I see no reason numbers are not added to the lexical grammar for table names. I am testing a fix that will let it start with numbers - but looks like you will alwayse have to include a letter in the name.

**Comment by sljohnsondev** [comment]
Thank you, Mathias!  I understand the constraint around numbers as table names and we can be sure to document the need to backtick the table name in that scenario. 

Appreciate the quick response!  Hope you all are doing well and staying safe and sane.

**Comment by sljohnsondev** [comment]
Mathias, one thing I just noticed when passing a number in as table name - i.e. `SELECT * FROM csv.50` - the parser is translating the from statement to a table value with alias (see below)

`yy.Table {
  "tableid": "csv",
  "as": ".50"
}`

Is this something that could be resolved to return an error or is this expected behavior?

**Comment by mathiasrw** [comment]
Interesting aspect. Thank you for letting me konw. 

I guess its the actual file name that is causing the problem. Have you tried to do a `SELECT * FROM CSV('filename')` ?

**Comment by sljohnsondev** [comment]
@mathiasrw sorry, that was a bad example.  csv in the above example is just the schema name that had been used in the bug reported to me.  It could be any string value - i.e. dev.50, temperature.101, etc.

**Comment by mathiasrw** [comment]
Fix released as part of v4.11.0
